src/evm/process-sponsor-wallet.ts

```diff
@@ -10,13 +11,31 @@ export const processSponsorWallet = async (
   contract: ethers.Contract,
   gasTarget: node.GasTarget,
   subscriptions: ProcessableSubscription[],
-  sponsorWallet: ethers.Wallet
+  sponsorWallet: ethers.Wallet,
+  voidSigner: ethers.VoidSigner
 ): Promise<node.LogsData<ProcessableSubscription>[]> => {
   const logs: node.LogsData<ProcessableSubscription>[] = [];
+  const sortedSubscriptions = subscriptions.sort((a, b) => a.nonce - b.nonce);
 
+  // Keep track of nonce outside of the loop in case there is an invalid subscription and its nonce is skipped
+  //TODO: improve nonce?
+  let nextNonce = sortedSubscriptions[0].nonce;
```

Comment:
I like that the nonces are not bound to the subscriptions and so we can skip invalid once. Let's see if there's some issue with that. But in case it works, we should maybe rename the field to something like `index` or `order` because the nonce doesn't have to be the nonce actually used which is misleading. Also, the (potentially) wrong nonce is logged.